src/main/java/com/faforever/server/ladder/DivisionService.java

```diff
@@ -0,0 +1,144 @@
+package com.faforever.server.ladder;
+
+import com.faforever.server.config.ServerProperties;
+import com.faforever.server.entity.Division;
+import com.faforever.server.entity.Player;
+import com.faforever.server.entity.PlayerDivisionInfo;
+import com.google.common.collect.ImmutableList;
+import lombok.Getter;
+import lombok.extern.slf4j.Slf4j;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+import org.springframework.stereotype.Service;
+
+import javax.annotation.PostConstruct;
+import javax.inject.Inject;
+import javax.transaction.Transactional;
+import java.text.MessageFormat;
+import java.util.Objects;
+
+@Service
+@Slf4j
+public class DivisionService {
+  private final ServerProperties properties;
+  private final DivisionRepository divisionRepository;
+  private final PlayerDivisionInfoRepository playerDivisionInfoRepository;
+
+  @Getter
+  private ImmutableList<Division> divisions;
+
+  @Inject
+  public DivisionService(ServerProperties properties, DivisionRepository divisionRepository, PlayerDivisionInfoRepository playerDivisionInfoRepository) {
+    this.properties = properties;
+    this.divisionRepository = divisionRepository;
+    this.playerDivisionInfoRepository = playerDivisionInfoRepository;
+  }
+
+  @PostConstruct
+  public void init() {
+    divisions = ImmutableList.copyOf(divisionRepository.findAllByOrderByLeagueAscThresholdAsc());
+  }
+
+  public Division getCurrentPlayerDivision(Player player) {
+    int season = properties.getLadder().getSeason();
+
+    PlayerDivisionInfo info = playerDivisionInfoRepository.findByPlayerAndSeason(player, season);
+
+    if(info == null) {
+      return null;
+    }
```

Comment:
You don't want the caller to deal with possible null values, so make the method return `Optional<Division>` instead.
And to always have the code correctly formatted, you can enable these in your commit dialog:
![image](https://user-images.githubusercontent.com/1864183/30424359-0256b628-9946-11e7-99a8-4442237f993a.png)